client/packages/system/src/Patient/Insurance/components/InsuranceModal/InsuranceModal.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { FC, ReactElement, useEffect, useState } from 'react';
+
+import { useDialog, useNotification, useUrlQuery } from '@common/hooks';
+import { DateUtils, useFormatDateTime, useTranslation } from '@common/intl';
+import {
+  BaseDatePickerInput,
+  BasicTextInput,
+  DialogButton,
+  InputWithLabelRow,
+  NumericTextInput,
+} from '@common/components';
+
+import {
+  Box,
+  FnUtils,
+  InsurancePolicyNodeType,
+  Stack,
+} from '@openmsupply-client/common';
+import {
+  PatientModal,
+  usePatientModalStore,
+} from '@openmsupply-client/programs';
+
+import { usePatient } from '../../../api';
+import { InsurancePolicySelect } from '../InsurancePolicySelect';
+import { InsuranceProvidersSelect } from '../InsuranceProvidersSelect';
+import { InsuranceStatusToggle } from '../InsuranceStatusToggle';
+
+const DEFAULT_INSURANCE = {
+  id: '',
+  policyNumberFamily: '',
+  policyNumberPerson: '',
+  insuranceProviderId: '',
+  policyType: '' as InsurancePolicyNodeType,
+  isActive: false,
+  discountPercentage: 0,
+  expiryDate: '',
+};
+
+export const InsuranceModal: FC = (): ReactElement => {
+  const t = useTranslation();
+  const formatDateTime = useFormatDateTime();
+  const { success, error } = useNotification();
+  const { current, setModal } = usePatientModalStore();
+
+  const { Modal } = useDialog({
+    disableBackdrop: true,
+    onClose: () => setModal(undefined),
+    isOpen: current === PatientModal.Insurance,
+  });
+
+  const nameId = usePatient.utils.id();
+  const { data } = usePatient.document.insurances({ nameId });
+  const [insurance, setInsurance] = useState(DEFAULT_INSURANCE);
```

Comment:
As a result of the newer hook patterns (mentioned below), this would also include having the Draft state in the hook, and this would be simpler here.